pontoon/tags/utils/tag.py

```diff
@@ -0,0 +1,118 @@
+
+from django.utils.functional import cached_property
+
+from .tagged import Tagged, TaggedLocale
+
+
+class TagTool(Tagged):
+    """This wraps ``Tag`` model kwargs providing an API for
+    efficient retrieval of related information, eg tagged ``Resources``,
+    ``Locales`` and ``Projects``, and methods for managing linked
+    ``Resources``.
+    """
+
+    def __init__(self, tags_tool, **kwargs):
+        self.tags_tool = tags_tool
+        self._name = kwargs.pop("name")
+        self._pk = kwargs.pop("pk")
+        self._priority = kwargs.pop("priority")
+        self._slug = kwargs.pop("slug")
+        self._project = kwargs.pop("project")
+        super(TagTool, self).__init__(**kwargs)
+
+    @property
+    def linkable_resources(self):
+        """``Resources`` that could be linked to this ``Tag``"""
+        return self.resource_tool.get_linkable_resources(
+            self.slug).order_by('path')
+
+    @property
+    def linked_resources(self):
+        """``Resources`` that are linked to this ``Tag``"""
+        return self.resource_tool.get_linked_resources(
+            self.slug).order_by('path')
+
+    @property
+    def locale_stats(self):
+        return self.tags_tool.stat_tool(
+            slug=self.slug,
+            groupby='locale').data
+
+    @property
+    def name(self):
+        """Name of ``Tag`` model object"""
+        return self._name
```

Comment:
What's the benefit of this property (and the 4 that follow below)?

Why not simply call them `self.name` etc. in the constructor method?

Reply:
agreed, ill update